src/opensynth/models/faraday/gaussian_mixture/fit_kmeans.py

```diff
@@ -8,76 +8,71 @@
 import pytorch_lightning as pl
 import torch
 from pytorch_lightning.loggers import CSVLogger
+from sklearn.cluster import KMeans
 from torch.utils.data import DataLoader
 
-from opensynth.models.faraday.gaussian_mixture.kmeans_lightning import (
-    KMeansLightningModule,
-    KmeansRandomInitLightningModule,
-)
-from opensynth.models.faraday.gaussian_mixture.model import KMeansModel
-
 logger = CSVLogger("lightning_logs", name="kmeans_logs")
 
 
 def fit_kmeans(
     data: DataLoader,
     num_components: int,
     vae_module: pl.LightningModule,
```

Comment:
You should type vae_module as an instance of FaradayVAE instead. The `reshape_data` method is a method of FaradayVAE class, not pl.LightningModule.

Reply:
Note that this has been fixed in another open PR (https://github.com/OpenSynth-energy/OpenSynth/pull/47)